Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -36,11 +37,11 @@ internal class HttpCoreDiagnosticSourceListener : IObserver<KeyValuePair<string,
         private readonly PropertyFetcher deprecatedResponseFetcher = new PropertyFetcher("Response");
         private readonly PropertyFetcher deprecatedRequestGuidFetcher = new PropertyFetcher("LoggingRequestId");
         private readonly PropertyFetcher deprecatedResponseGuidFetcher = new PropertyFetcher("LoggingRequestId");
-        
+
         #endregion
 
-        private readonly ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>> pendingTelemetry =
-            new ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>>();
+        private readonly ConditionalWeakTable<HttpRequestMessage, IOperationHolder<DependencyTelemetry>> pendingTelemetry = 
```

Comment:
WE call StopOperation that [disposes](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/src/Core/Managed/Shared/TelemetryClientExtensions.cs#L178) operation holder and it [tracks](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/src/Core/Managed/Shared/Extensibility/Implementation/OperationHolder.cs#L110) operation.

Oh, but it would mean that dependency will be tracked anyway when the HttpRequestMessage (the key) is disposed, which is a good side effect